File: 0013-roman-to-integer/0013-roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
        HashMap<Character, Integer> m = new HashMap<>();
        m.put('I', 1);
        m.put('V', 5);
        m.put('X', 10);
        m.put('L', 50);
        m.put('C', 100);
        m.put('D', 500);
        m.put('M', 1000);
        int n=s.length();
        int ans=0;
        for(int i=0;i<n;i++){
            int curval=m.get(s.charAt(i));
            if(i+1 < n && curval < m.get(s.charAt(i+1))){
                int totalval=m.get(s.charAt(i+1));
                ans+=(totalval-curval);
                i++;
            }else{
                ans+=curval;
            }
        }
        return ans;
    }
}

File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        m ={
            "I" : 1,
            "V" : 5,
            "X" : 10,
            "L" : 50,
            "C" : 100,
            "D" : 500,
            "M" : 1000
        }
        ans = 0
        i = 0
        while i < len(s):
            cur = m[s[i]]
            if i+1 < len(s) and cur < m[s[i+1]]:
                nextval = m[s[i+1]]
                totalval = nextval - cur
                ans+=totalval
                i+=1
            else:
                ans+=cur
            i+=1
        return ans
        